src/types.rs

```diff
@@ -914,7 +915,8 @@ fn push_four_digits(dest: &mut WriteBuf, val: u16) -> WriteResult {
 }
 
 /// A structure representing a (UTC timezone) date and time.
-/// Wrapped by `UtcTime` and `X509GeneralizedTime`.
+/// Wrapped by `UtcTime` and `X509GeneralizedTime` and used in
+/// `GeneralizedTime`.
 #[derive(Debug, Clone, PartialEq, Hash, Eq, PartialOrd)]
 pub struct DateTime {
```

Comment:
Should DateTime just have an optional nanoseconds field?

Reply:
Thoughts here?

Reply:
Whoops, missed this earlier. I don't have a super strong opinion -- I think @DarkaMaul put the nanos on `GeneralizedTime` since `UTCTime` has no fractional time support at all, so having it on the top-level `DateTime` used by both seems (slightly) off. 

OTOH having it on `GeneralizedTime` also seems slightly off, since it's now `(DateTime, nanos)`. So I'm happy to move if you'd prefer.

Reply:
Eh, let's get the docs right on the generalized tiem strcut and that'll be good enough. (See my other comment)